pcdet/models/detectors/pv_rcnn_ssl.py

```diff
@@ -320,18 +320,18 @@ def forward(self, batch_dict):
     # TODO(farzad) Warning: this method has two types of stats: one maintaining their stats over batches, e.g., TP, FP
     #  and two reset state after each batch. Use update_metrics flag if you want call the method twice in different
     #  places during a batch.
-    def calc_statistics(self, pred_boxes, gt_boxes, update_metrics=True, tag=None, **kwargs):
-        assert len(pred_boxes) == len(gt_boxes)
-
+    def calc_statistics(self, pred_boxes_org, gt_boxes_org, update_metrics=True, tag=None, **kwargs):
+        assert len(pred_boxes_org) == len(gt_boxes_org)
+        
         statistics = defaultdict(list)
-        for i in range(len(pred_boxes)):
-            valid_preds_mask = pred_boxes[i].sum(dim=-1) > 0
```

Comment:
It turns out that this is not the correct way to filter empty (all-zero) rows, as the sum over columns could sometimes be negative!

Reply:
Probably it can be `valid_preds_mask = pred_boxes[i].sum(dim=-1) != 0`.

Reply:
Using `valid_preds_mask = pred_boxes[i].sum(dim=-1) != 0` , we are summing all the elements of each `pred_boxes`. 

Eg. 
pred_boxes[i] = tensor([[  5.1124, -15.7919,  -0.7340,   4.9734,   2.0533,   2.1049,  -0.5451, 1.0000],
        [ 10.4329,   2.0733,  -0.6736,   4.7155,   1.8057,   1.7954,   1.0749, 1.0000],
        [ 15.8639,  -1.3333,  -0.6113,   4.1480,   1.7335,   1.7748,  -2.1551, 1.0000], device='cuda:0')

`pred_boxes[i].sum(dim=-1)` = [-1.8269, 22.2240, 20.4205]

But there could be a case when this sum is zero for a box, whereas the elements of the corresponding pred box are actually non-zero. To avoid that case, can we use `valid_preds_mask = pred_boxes[i][:, -1] != 0`

Reply:
Yeah I was thinking about the case when the sum can be zero even though the element,s are non-zero but it's really unlikely. That's why I said this is "probably" correct. I was also thinking about `valid_preds_mask = pred_boxes[i][:, -1] != 0` option. I think it's likely that the class labels are set to zero somewhere in the code or in the future. So checking this only based on the label might be risky. Having said that, a good way is maybe to check exactly for zero and reduce over `all` columns.